### PR TITLE
TINY-9251 & TINY-10259: Move build to containers and replace phantomjs with chrome-headless

### PR DESCRIPTION
**Pull request by jscasca**
Related Ticket: TINY-9251, TINY-10259

Description of Changes:
* Move build to containers to satisfy Node >= 16 requirement of https://github.com/tinymce/tinymce/pull/9049
* Replace phantomjs with chrome-headless
* Fix test failures - previously these tests were skipped when testing on PhantomJS so the issues did not surface
* Remove PhantomJS skips in tests
* Remove PhantomJS mentions in README
* Replace PhantomJS scripts
* Some follow-up tasks around phantomjs to chrome-headless migration to be done in TINY-10265

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] ~Milestone set~
* [x] ~Docs ticket created (if applicable)~

GitHub issues (if applicable):

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -35,44 +39,45 @@ def runBrowserTests(name, browser, os, bucket, buckets, runAll) {
...
-  timestamps {
-    checkout scm
-
-    def props = readProperties file: 'build.properties'
-
-    def primaryBranch = props.primaryBranch
+timestamps {
+  // TinyMCE builds need more CPU and RAM (especially eslint)
+  // NOTE: Ensure not to go over 7.5 CPU/RAM due to EC2 node sizes and the jnlp container requirements
+  tinyPods.node([
```
Comment (ltrouton) [inline Jenkinsfile:57]: On `develop` and `master` we are using `tinyPods.nodeBrowser`. Just wanted to check if what we have here is fine
```suggestion
  ttinyPods.nodeBrowser([
```

**Review by ltrouton — APPROVED** [review]
I am wondering if we need to note something in the changelog about switching to chrome-headless for tests. I am not sure as it is not user-facing but developer-facing and technically could affect them

**Comment by mkzhx** [comment]
> I am wondering if we need to note something in the changelog about switching to chrome-headless for tests. I am not sure as it is not user-facing but developer-facing and technically could affect them

@ltrouton The task to replace phantomjs with chrome-headless for TinyMCE 6 wasn't documented in the changelog so I reckon let's keep it consistent. The change is documented in CONTRIBUTING.md which is more developer-facing.

File: Jenkinsfile (hunk 1 above)
Comment (jscasca) [reply Jenkinsfile:57]: @mkzhx this should run a selenium chrome container to run the headless tests. Recommended

Comment (mkzhx) [reply Jenkinsfile]: Thanks guys, suggestion applied.

**Review by shanmen-tiny — APPROVED** [review]

**Review by tiny-ben-tran — APPROVED** [review]

**Review by tiny-james — APPROVED** [review]

**Comment by TheSpyder** [comment]
@mkzhx @ltrouton yes no need to note in the changelog. It doesn't matter how tests are run, only that they are

**Review by TheSpyder — APPROVED** [review]

**Pull request merged**